src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ThemeService.cs

```diff
@@ -0,0 +1,147 @@
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using static GovUk.Education.ExploreEducationStatistics.Admin.Validators.ValidationUtils;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class ThemeService : IThemeService
+    {
+        private readonly ContentDbContext _context;
+        private readonly IMapper _mapper;
+        private readonly IPersistenceHelper<ContentDbContext> _persistenceHelper;
+        private readonly IUserService _userService;
+
+        public ThemeService(
+            ContentDbContext context,
+            IMapper mapper,
+            IUserService userService,
+            IPersistenceHelper<ContentDbContext> persistenceHelper)
+        {
+            _context = context;
+            _mapper = mapper;
+            _userService = userService;
+            _persistenceHelper = persistenceHelper;
+        }
+
+        public async Task<Either<ActionResult, ThemeViewModel>> CreateTheme(SaveThemeViewModel createdTheme)
+        {
+            return await _userService.CheckCanManageAllTaxonomy()
+                .OnSuccess(
+                    async _ =>
+                    {
+                        if (_context.Themes.Any(theme => theme.Slug == createdTheme.Slug))
+                        {
+                            return ValidationActionResult(ValidationErrorMessages.SlugNotUnique);
+                        }
+
+                        var saved = await _context.Themes.AddAsync(
+                            new Theme
+                            {
+                                Slug = createdTheme.Slug,
+                                Summary = createdTheme.Summary,
+                                Title = createdTheme.Title,
+                            }
+                        );
+
+                        await _context.SaveChangesAsync();
+                        return await GetTheme(saved.Entity.Id);
+                    }
+                );
+        }
+
+        public async Task<Either<ActionResult, ThemeViewModel>> UpdateTheme(
+            Guid themeId,
+            SaveThemeViewModel updatedTheme)
+        {
+            return await _persistenceHelper.CheckEntityExists<Theme>(themeId)
+                .OnSuccessDo(_userService.CheckCanManageAllTaxonomy)
+                .OnSuccess(
+                    async theme =>
+                    {
+                        if (_context.Themes.Any(t => t.Slug == updatedTheme.Slug && t.Id != themeId))
+                        {
+                            return ValidationActionResult(ValidationErrorMessages.SlugNotUnique);
```

Comment:
I'm just wondering if we even do anything with Theme and Topic slugs. I can't see where they are used for navigation so I'm wondering if we even need them.

Reply:
They're not being used but think we've agreed that we'll keep this in for the time being in case we need them in the future. They provide uniqueness validation anyway which is useful.